### ISSUE TITLE
Error writing data to identity/entity

### ISSUE THREAD

**Issue by amoravek** [hashicorp/vault#12589]
**Describe the bug**
The command:

vault write identity/entity name="bob-smith" policies="base"  metadata=organization="ACME Inc." metadata=team="QA"

shuts down a fresh vault installation deployed via Helm chart (hashicorp/vault v0.15.0) on ARM64 Kubernetes cluster.

**To Reproduce**
1. Install vault on ARM64 hardware
2. unseal and login using root token
3. Run: vault write identity/entity name="bob-smith" policies="base"  metadata=organization="ACME Inc." metadata=team="QA"
4. Vault CLI produces error: Error writing data to identity/entity: Put "http://xx.xx.xx.xx:8200/v1/identity/entity": EOF
5. Vault crashes, produces error stacktrace and seals itself

**Expected behavior**
entity is successfully created without any error

**Environment:**
* Vault Server Version 1.8.1 or 1.8.2
* Vault CLI Version (retrieve with `vault version`): Vault v1.8.2 (aca76f63357041a43b49f3e8c11d67358496959f)
* Server Operating System/Architecture: Ubuntu 20.04.3 LTS, ARM64, kernel 4.4.154-113-rockchip-gdb9dfc2cdd25

Vault server configuration file(s):

```hcl
disable_mlock = true
ui = true

listener "tcp" {
  tls_disable = 1
  address = "[::]:8200"
  cluster_address = "[::]:8201"
}
storage "file" {
  path = "/vault/data"
}

# Example configuration for using auto-unseal, using Google Cloud KMS. The
# GKMS keys must already exist, and the cluster must have a service account
# that is authorized to access GCP KMS.
#seal "gcpckms" {
#   project     = "vault-helm-dev"
#   region      = "global"
#   key_ring    = "vault-helm-unseal-kr"
#   crypto_key  = "vault-helm-unseal-key"
#}
```
[vault1-vault1-0-1632176223949848366.log](https://github.com/hashicorp/vault/files/7199031/vault1-vault1-0-1632176223949848366.log)

**Additional context**

**Comment by amoravek** [comment]
...tried the same steps also in DEV mode. Result was the same.

**Comment by ncabatoff** [comment]
Hi @amoravek,

Are you sure about the version?  We fixed this issue when it was reported as https://github.com/hashicorp/vault/issues/12368, and the fix should be present in 1.8.2.  Also, the logs you attach show v1.8.1 was the version that panicked.

**Comment by amoravek** [comment]
You are right - I've found out that the 1.8.2 upgrade was automatically rolled-back by my ArgoCD instance configured to automated sync policy  (without me noticing). I've tried to install the newest helm chart 0.16.0 with vault 1.8.2 and the problem is gone. So the 1.8.2 version fixed it for me. Sorry about that.
